docs/reference-guides/interactivity-api/api-reference.md

```diff
@@ -325,6 +325,11 @@ The `wp-on` directive is executed each time the associated event is triggered.
 
 The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored.
 
+### `wp-on-async`
+
+This directive is a more performant approach of `wp-on`. The action will be yielded to the main thread to not block it, allowing other interactions that otherwise would be waiting on the main thread
+to run sooner. Use this directive for any `wp-on` that doesn't need the `event` object (`event.preventDefault()`, `event.stopPropagation()`, etc.)
```

Comment:
```suggestion
to run sooner. Use this directive for any `wp-on` that doesn't need synchronous access to the `event` object, in particular the methods `event.preventDefault()`, `event.stopPropagation()`, and `event.stopImmediatePropagation()`.
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/62663/commits/e3b049796f2ef52635168d775e967d2cd7cb338f